File: weather-library/src/main/java/org/rcgonzalezf/weather/openweather/network/OpenWeatherApiError.java
package org.rcgonzalezf.weather.openweather.network;

import org.rcgonzalezf.weather.common.network.ApiError;

class OpenWeatherApiError implements ApiError<String>{

  private String mMessage;
  private ErrorCode mCode;

  public void setMessage(String message) {
    mMessage = message;
  }

  @Override
  public String getError() {
    return mMessage;
  }

  @Override
  public void setCode(ErrorCode code) {
    mCode = code;
  }
}

File: weather-library/src/main/java/org/rcgonzalezf/weather/openweather/network/OpenWeatherApiError.kt
package org.rcgonzalezf.weather.openweather.network

import org.rcgonzalezf.weather.common.network.ApiError

class OpenWeatherApiError : ApiError<String?> {
    override var error: String? = null
        private set
    private var code: ErrorCode? = null
    fun setMessage(message: String?) {
        error = message
    }

    override fun setCode(code: ErrorCode?) {
        this.code = code
    }

    fun getCode(): ErrorCode? {
        return code
    }
}
